deviceModeInit.js

```diff
@@ -233,35 +283,33 @@ var rudderTracking = (function () {
     const timeToUpdate = thresholdTime - (currentTime - last_updated_at);
     return timeToUpdate
   }
-
-  function sendIdentifierToRudderWebhook(cart) {
+  function sendToRudderWebhook(data, type, updateTypeCookieFunction, maxRetries = 3) {
     const webhookUrl =
       "https://dataplaneUrl_placeHolder/v1/webhook?writeKey=writeKey_placeHolder";
-    const data = {
-      event: "rudderIdentifier",
-      anonymousId: rudderanalytics.getAnonymousId(),
-      cartToken: cart.token,
-      cart: cart
-    };
-    rs$
-      .ajax({
-        url: webhookUrl,
-        method: "POST",
-        contentType: "application/json",
-        data: JSON.stringify(data),
-      })
-      .then(() => {
-        updateTimeStampForIdentifierEvent();
-        console.log("Successfully sent identifier event to rudderstack");
-      })
-      .catch(() => {
-        console.log("Failed to sent identifier event to rudderstack");
-      });
+    var timeToRetry = 1000; // 1 second
+    if (maxRetries > 0) {
+      rs$
+        .ajax({
+          url: webhookUrl,
+          method: "POST",
+          contentType: "application/json",
+          data: JSON.stringify(data),
+        })
+        .then(() => {
+          updateTypeCookieFunction();
+          console.log(`Successfully sent ${type} event to rudderstack`);
+        })
+        .catch(() => {
+          setTimeout(() => { sendToRudderWebhook(data, type, updateTypeCookieFunction, maxRetries - 1) }, timeToRetry);
```

Comment:
we should use exponential backoff, timeToRetry should be multiple of of retryCount

Reply:
Done